### PR TITLE
Fixed caching rules in  with different ruleConfig configuration that …

### PR DESCRIPTION
**Pull request by dmirogin**
Fixed caching rules in `yii\web\UrlManager` with different ruleConfig configuration

| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #14406

### CODE REVIEW

File: tests/framework/web/UrlManagerCreateUrlTest.php [hunk 1]
```diff
@@ -758,4 +759,34 @@ public function testUrlCreateCacheWithParameterMismatch()
...
     }
+
+    /**
+     * @see https://github.com/yiisoft/yii2/issues/14406
+     */
+    public function testCreatingRulesWithDifferentRuleConfigAndEnabledCache()
+    {
+        $this->mockWebApplication([
+            'components' => [
+                'cache' => ArrayCache::class
```
Comment (rob006) [inline tests/framework/web/UrlManagerCreateUrlTest.php]: You can not use `::class` - it will now work on PHP 5.4.

Comment (dmirogin) [inline tests/framework/web/UrlManagerCreateUrlTest.php]: @rob006 Thanks!

**Review by samdark — APPROVED** [review]

**Review by cebe — APPROVED** [review]

File: tests/framework/web/UrlManagerCreateUrlTest.php [hunk 2]
```diff
@@ -758,4 +759,34 @@ public function testUrlCreateCacheWithParameterMismatch()
...
+     * @see https://github.com/yiisoft/yii2/issues/14406
+     */
+    public function testCreatingRulesWithDifferentRuleConfigAndEnabledCache()
+    {
+        $this->mockWebApplication([
+            'components' => [
+                'cache' => ArrayCache::className()
+            ]
+        ]);
+        $urlManager = $this->getUrlManager([
```
Comment (dynasource) [inline tests/framework/web/UrlManagerCreateUrlTest.php]: $urlManager does not seem to be used and is therefore hard to understand. Perhabs can you add an assertion similar to what you  do for $cachedUrlManager

File: tests/framework/web/UrlManagerCreateUrlTest.php [hunk 3]
```diff
@@ -758,4 +759,34 @@ public function testUrlCreateCacheWithParameterMismatch()
...
+            'cache' => 'cache',
+            'ruleConfig' => [
+                'class' => CachedUrlRule::className()
+            ],
+            'rules' => [
+                '/' => 'site/index'
+            ]
+        ]);
+
+        $this->assertInstanceOf(CachedUrlRule::className(), $cachedUrlManager->rules[0]);
```
Comment (dynasource) [inline tests/framework/web/UrlManagerCreateUrlTest.php]: it would be more logical to verify that this $rule is not similar to that of $urlManager

**Review by dynasource — CHANGES_REQUESTED** [review]

**Comment by dmirogin** [comment]
Made tests clearer.

**Review by dynasource — APPROVED** [review]
thx

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**